(Master) the name `_AssertComponentSomeComponent` is defined multiple times

I'm trying to do the following
```rust
#[openapi
(
    ...
    components(
        handlers::some_mod::another_one::one_last::request::body::SomeComponent as request::body::SomeComponent,
        handlers::some_mod::another_one::one_last::response::SomeComponent as response::SomeComponent,
    )
)]
```
but this gives me the following error
```rust
error[E0428]: the name `_AssertComponentSomeComponent` is defined multiple times
   --> src/main.rs:215:89
    |
214 | ...ast::request::body::SomeComponent as request::body::SomeComponent,
    |                         -------------------- previous definition of the type `_AssertComponentSomeComponent` here
215 | ...ast::response::SomeComponent as response::SomeComponent,
    |                    ^^^^^^^^^^^^^^^^^^^^ `_AssertComponentSomeComponent` redefined here
    |
    = note: `_AssertComponentSomeComponent` must be defined only once in the type namespace of this block
```
I thought https://github.com/juhaku/utoipa/pull/187 solved this issue?

--- a/scripts/test.sh
+++ b/scripts/test.sh
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+set -e
+
 crate="$1"
 
 echo "Testing crate: $crate..."

--- a/utoipa-gen/src/openapi.rs
+++ b/utoipa-gen/src/openapi.rs
@@ -225,20 +225,12 @@ impl ToTokens for OpenApi {
         let OpenApi(attributes, ident) = self;
 
         let info = info::impl_info();
-        let components = impl_components(&attributes.components, tokens).map(|components| {
+        let components = impl_components(&attributes.components).map(|components| {
             quote! { .components(Some(#components)) }
         });
 
         let modifiers = &attributes.modifiers;
         let modifiers_len = modifiers.len();
-
-        modifiers.iter().for_each(|modifier| {
-            let assert_modifier = format_ident!("_Assert{}", modifier.ident);
-            let ident = &modifier.ident;
-            quote_spanned! {modifier.ident.span()=>
-                struct #assert_modifier where #ident : utoipa::Modify;
-            };
-        });
 
         let path_items = impl_paths(&attributes.handlers);
 
@@ -280,18 +272,14 @@ impl ToTokens for OpenApi {
     }
 }
 
-fn impl_components(
-    components: &Punctuated<Component, Comma>,
-    tokens: &mut TokenStream,
-) -> Option<TokenStream> {
+fn impl_components(components: &Punctuated<Component, Comma>) -> Option<TokenStream> {
     if !components.is_empty() {
         let mut components_tokens = components.iter().fold(
             quote! { utoipa::openapi::ComponentsBuilder::new() },
             |mut schema, component| {
                 let path = &component.path;
                 let ident = component.get_ident().unwrap();
-                let span = ident.span();
-                // let component_name: String = ident.to_string();
+
                 let component_name: String = component
                     .alias
                     .as_ref()
@@ -300,23 +288,13 @@ fn impl_components(
 
                 let (_, ty_generics, _) = component.generics.split_for_impl();
 
-                let assert_ty_generics = if component.has_lifetime_generics() {
-                    Some(quote! {<'static>})
-                } else {
-                    Some(ty_generics.to_token_stream())
-                };
-                let assert_component = format_ident!("_AssertComponent{}", ident);
-                tokens.extend(quote_spanned! {span=>
-                    struct #assert_component where #path #assert_ty_generics: utoipa::Component;
-                });
-
                 let ty_generics = if component.has_lifetime_generics() {
                     None
                 } else {
                     Some(ty_generics)
                 };
 
-                schema.extend(quote! {
+                schema.extend(quote_spanned! {ident.span()=>
                     .component(#component_name, <#path #ty_generics>::component())
                     .components_from_iter(<#path #ty_generics>::aliases())
                 });

--- a/utoipa-gen/src/path/property.rs
+++ b/utoipa-gen/src/path/property.rs
@@ -1,4 +1,4 @@
-use quote::{format_ident, quote, quote_spanned, ToTokens};
+use quote::{quote, quote_spanned, ToTokens};
 use syn::spanned::Spanned;
 
 use crate::{
@@ -50,12 +50,8 @@ impl ToTokens for Property<'_> {
             let name = component::format_path_ref(component_name_path);
 
             let component = if self.0.is_inline {
-                let assert_component = format_ident!("_Assert{}", name);
-                quote_spanned! { component_name_path.span() => {
-                        struct #assert_component where #component_name_path: utoipa::Component;
-
-                        <#component_name_path as utoipa::Component>::component()
-                    }
+                quote_spanned! { component_name_path.span()=>
+                    <#component_name_path as utoipa::Component>::component()
                 }
             } else {
                 quote! {

